src/libcore/iter.rs

```diff
@@ -1273,60 +1269,21 @@ impl<T: Clone> MinMaxResult<T> {
 /// An iterator that clones the elements of an underlying iterator
 #[unstable(feature = "core", reason = "recent addition")]
 #[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
-#[derive(Clone)]
-pub struct Cloned<I> {
```

Comment:
Using a newtype structure which is opaque to its contents is an intentional decision on behalf of `std::iter`. It allows us to flexibly change the representation in the future if necessary.